### PR TITLE
Improve bench time by minimizing docker container creation

### PR DESCRIPTION
**Pull request by rukai**
Uses the added new_lazy_shared function to lazily create a single BenchResources that is shared by every benchmark that needs the same resources.

There is of course a lot of variance between runs but 32m looks pretty good.

### CODE REVIEW

File: shotover-proxy/benches/cassandra_benches.rs [hunk 1]
```diff
@@ -105,50 +134,64 @@ fn cassandra(c: &mut Criterion) {
...
+                        connection.execute(&query.statement).wait().unwrap();
                     })
                 },
             );
         }
     }
 }
 
+// This can be easily pulled out and reused by other benchmarks if needed.
+fn new_lazy_shared<T>(create: impl Fn() -> T) -> impl Fn() -> Arc<RefCell<Option<T>>> {
```
Comment (conorbros) [inline shotover-proxy/benches/cassandra_benches.rs:185]: Mind adding a comment how this works?

Comment (rukai) [reply shotover-proxy/benches/cassandra_benches.rs:190]: yeah, this function touches a whole bunch of weird corners of rust.
I added some comments explaining what it does, let me know if you have any questions.

Comment (conorbros) [reply shotover-proxy/benches/cassandra_benches.rs:192]: Yep there is a bit of magic in there. Thanks!

**Review by conorbros — APPROVED** [review]

**Review by benbromhead — APPROVED** [review]

**Comment by benbromhead** [comment]
noice

**Comment by rukai** [comment]
rebased

**Pull request merged**